plugin/trino-redshift/src/test/java/io/trino/plugin/redshift/TestRedshiftCastPushdown.java

```diff
@@ -393,6 +422,39 @@ protected List<CastTestCase> supportedCastTypePushdown()
                 new CastTestCase("c_numeric_30_2", "integer", "c_integer"),
                 new CastTestCase("c_decimal_negative", "integer", "c_integer"),
 
+                new CastTestCase("c_char_10", "char(50)", "c_char_50"),
+                new CastTestCase("c_char_10", "char(256)", "c_bpchar"),
+                new CastTestCase("c_varchar_10", "varchar(50)", "c_varchar_50"),
+                new CastTestCase("c_nvarchar_10", "varchar(50)", "c_nvarchar_50"),
+                new CastTestCase("c_varchar_10", "varchar(50)", "c_text"),
+
+                new CastTestCase("c_char_50", "char(10)", "c_char_10"),
+                new CastTestCase("c_bpchar", "char(10)", "c_char_10"),
+                new CastTestCase("c_varchar_50", "varchar(10)", "c_varchar_10"),
+                new CastTestCase("c_nvarchar_50", "varchar(10)", "c_nvarchar_10"),
+                new CastTestCase("c_text", "varchar(10)", "c_varchar_10"),
+
+                new CastTestCase("c_char_10", "char(50)", "c_char_50"),
+                new CastTestCase("c_char_10", "char(256)", "c_bpchar"),
+                new CastTestCase("c_char", "char(4096)", "c_char_4096"),
+                new CastTestCase("c_char", "char(1)", "c_nchar"),
+                new CastTestCase("c_varchar_10", "varchar(50)", "c_varchar_50"),
+                new CastTestCase("c_nvarchar_10", "varchar(50)", "c_nvarchar_50"),
+                new CastTestCase("c_varchar_10", "varchar(50)", "c_text"),
+
+                new CastTestCase("c_varchar_50", "varchar(10)", "c_varchar_10"),
+
+                new CastTestCase("c_varchar_10", "varchar(50)", "c_varchar_50"),
+                new CastTestCase("c_varchar_10", "varchar(65535)", "c_varchar_65535"),
+                new CastTestCase("c_varchar_10", "varchar(256)", "c_nvarchar"),
+                new CastTestCase("c_varchar_10", "varchar(10)", "c_nvarchar_10"),
+                new CastTestCase("c_varchar_10", "varchar(50)", "c_nvarchar_50"),
+                new CastTestCase("c_varchar_10", "varchar(65535)", "c_nvarchar_65535"),
+                new CastTestCase("c_varchar_10", "varchar(256)", "c_text"),
+
+                new CastTestCase("c_varchar_unicode", "varchar(50)", "c_varchar_50"),
+                new CastTestCase("c_nvarchar_unicode", "varchar(50)", "c_varchar_50"),
```

Comment:
No, it supports max precision up to 65535 https://docs.aws.amazon.com/redshift/latest/dg/r_Character_types.html

Reply:
In `CastTestCase`, `castType` means trino type. When casting to unbounded varchar I see failure. Could you please check on this?

```
io.trino.testing.QueryFailedException: Type mismatch: column handle has type varchar but JdbcTypeHandle[jdbcType=12, jdbcTypeName=Optional[varchar], columnSize=Optional.empty, decimalDigits=Optional.empty, arrayDimensions=Optional.empty, caseSensitivity=Optional.empty] is mapped to varchar(65535)
```

```
        getQueryRunner().execute("CREATE TABLE test (a varchar(100))");
        getQueryRunner().execute("INSERT INTO test values ('Hello')");
        getQueryRunner().execute("SELECT cast (a as varchar) FROM test");
```

Reply:
thanks for explicitly asking for adding the test with unbounded types. I had this working until recently when I changed the default precision to REDSHIFT_MAX_VARCHAR instead of VarcharType.MAX_LENGTH(which I had before)

https://github.com/trinodb/trino/commit/b742602746a9ebe33c5641f89975618d4c3d78cf#diff-e7f2966e1e2e070e8643b5563c7ba62f7dc0652f76fbf446ef389b75ef4883fbL669

added below tests under supported.
```java
new CastTestCase("c_varchar_10", "varchar", "c_text"),
new CastTestCase("c_varchar_unicode", "varchar", "c_varchar_50"),
```